utils/kafka/src/main/java/io/apicurio/registry/utils/common/ConcurrentUtil.java

```diff
@@ -0,0 +1,89 @@
+package io.apicurio.registry.utils.common;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * @author Ales Justin
+ */
+public class ConcurrentUtil {
+
+    public static <T> T get(CompletableFuture<T> cf) {
+        boolean interrupted = false;
+        try {
+            while (true) {
+                try {
+                    return cf.get();
+                } catch (InterruptedException e) {
+                    interrupted = true;
```

Comment:
I hate `InterruptedException`; I often find it difficult to deal with. What you're doing here is essentially ignoring all interruptions and waiting a potentially arbitrary amount of time (until `cf` is completed) before exiting the loop and resetting the interrupted state. That's not really reacting to the interruption in a timely way. Are we expecting this thread (and TBH I'm not clear on whose thread this is, a Kafka one?) to ever be interrupted? If not it might be better to just throw the InterruptedException and let the caller, `StreamsRegistryStorage`, handle it by throwing `RegistryStorageException("Unexpected interruption")`, or something.

Reply:
@alesj Any thoughts on this before I merge?

Reply:
Let's leave it for now -- I can always remove this loop and just throw this fwd.
Where this code also unwraps ExecutionException.

Reply:
Ah, OK, now I remember why we handle CF::get in such a way -- this is an old pattern ...
The problem is that InterruptedException is checked, hence to propagate it all the way, all methods would have to have it (or plain Exception) in the signature == super ugly.
Hence the way to propagate this info is to silently ignore it, but do set the interrupt status back on the thread -- so that any cleanup code is potentially aware of it.